*** zerodf.c.orig	Wed Mar  1 16:42:23 2000
--- zerodf.c	Wed Mar  1 16:39:10 2000
***************
*** 130,136 ****
      cc  = c  + m + 1;
      inc = cc + m + 1;
      d   = inc+ m + 1;
!     
      if(freadf(c, sizeof(*c), m+1, fpc) != m+1) exit(1);
  
      for(;;){
--- 130,136 ----
      cc  = c  + m + 1;
      inc = cc + m + 1;
      d   = inc+ m + 1;
!  
      if(freadf(c, sizeof(*c), m+1, fpc) != m+1) exit(1);
  
      for(;;){
***************
*** 142,151 ****
  	for(j=fprd, i=(iprd+1)/2; j--;){
  	    if (freadf(&x, sizeof(x), 1, fp) != 1) exit(0);
  
! 	    if (! tp)
! 		x = zerodf(x, c, m, d, ngain);
  	    else
! 		x = zerodft(x, c, m, d, ngain);
  	    
  	    fwritef(&x, sizeof(x), 1, stdout);
  			
--- 142,154 ----
  	for(j=fprd, i=(iprd+1)/2; j--;){
  	    if (freadf(&x, sizeof(x), 1, fp) != 1) exit(0);
  
! 	    if (ngain)
! 		for(i=m; i>=0; i--) c[i] /= c[0];
! 
! 	    if (!tp)
! 		x = zerodf(x, c, m, d);
  	    else
! 		x = zerodft(x, c, m, d);
  	    
  	    fwritef(&x, sizeof(x), 1, stdout);
  			
